scipy/sparse/_sputils.py

```diff
@@ -297,7 +302,8 @@ def validateaxis(axis) -> None:
         raise ValueError("axis out of range")
 
 
-def check_shape(args, current_shape=None, *, allow_1d=False) -> tuple[int, ...]:
+def check_shape(args, current_shape=None, *, allow_1d=False,
+                allow_nd=False) -> tuple[int, ...]:
```

Comment:
minor: please feel free to delete ` -> tuple[int, ...]:`. Having only half a type annotation in a signature is not right.